File: eco-core/core-proxy/src/main/java/com/willfp/eco/proxy/proxies/VillagerTradeProxy.java
package com.willfp.eco.proxy.proxies;

import com.willfp.eco.core.proxy.AbstractProxy;
import org.bukkit.inventory.MerchantRecipe;
import org.jetbrains.annotations.NotNull;

public interface VillagerTradeProxy extends AbstractProxy {
    /**
     * Display a MerchantRecipe.
     *
     * @param recipe The recipe.
     * @return The new recipe.
     */
    MerchantRecipe displayTrade(@NotNull MerchantRecipe recipe);
}

File: eco-core/core-proxy/src/main/kotlin/com/willfp/eco/internal/spigot/proxy/VillagerTradeProxy.kt
package com.willfp.eco.internal.spigot.proxy

import org.bukkit.entity.Player
import org.bukkit.inventory.MerchantRecipe

interface VillagerTradeProxy {
    fun displayTrade(
        recipe: MerchantRecipe,
        player: Player
    ): MerchantRecipe
}